src/ZoneServer/Skills/Combat/HitInfo.cs

```diff
@@ -87,14 +101,18 @@ public HitInfo(ICombatEntity attacker, ICombatEntity target, Skill skill, float
 		/// <param name="skillId"></param>
 		/// <param name="damage"></param>
 		/// <param name="resultType"></param>
-		public HitInfo(ICombatEntity attacker, ICombatEntity target, SkillId skillId, float damage, HitResultType resultType)
+		///	<param name="hitCount"></param>
+		///	<param name="damageDelay"></param>
+		public HitInfo(ICombatEntity attacker, ICombatEntity target, SkillId skillId, float damage, HitResultType resultType, int hitCount = 0, TimeSpan damageDelay = default)
 		{
 			this.Attacker = attacker;
 			this.Target = target;
 			this.SkillId = skillId;
 
 			this.Damage = damage;
 			this.ResultType = resultType;
+			this.HitCount = hitCount;
+			this.DamageDelay = damageDelay == default ? TimeSpan.Zero : damageDelay;
```

Comment:
This is supposed to check if the damangeDelay argument has been set on the function call.
In this case the damageDelay will be set to TimeSpam.Zero.
This is needed since we CAN'T initialize the argument with TimeSpan.Zero (compilation error).

Reply:
Ohhh I think I understand it. Nevermind!

Reply:
No, it's not needed because the default value of `TimeSpan` is `TimeSpan.Zero`^^
```cs
default(TimeSpan) == TimeSpan.Zero
```
That means what you're checking here is effectively this.
```cs
this.DamageDelay = damageDelay == TimeSpan.Zero ? TimeSpan.Zero : damageDelay;
```
Meaning: If delay is Zero, then set the property to Zero. If it's not, set it to the delay. It's comparable to the following.
```cs
var value = var == 10 ? 10 : var;
```
It would be different if `TimeSpan` were a reference type, with a default value of null, but since it's a struct, and its default value is `TimeSpan.Zero`, the check is not needed and you can just do this.
```cs
this.DamageDelay = damageDelay;
```